save-backend/src/main/kotlin/org/cqfn/save/backend/service/TestExecutionService.kt

```diff
@@ -25,10 +28,13 @@ class TestExecutionService(private val testExecutionRepository: TestExecutionRep
                            private val testRepository: TestRepository,
                            private val agentRepository: AgentRepository,
                            private val executionRepository: ExecutionRepository,
-                           private val testDataFilesystemRepository: TestDataFilesystemRepository,
+                           transactionManager: PlatformTransactionManager,
 ) {
     private val log = LoggerFactory.getLogger(TestExecutionService::class.java)
     private val locks: ConcurrentHashMap<Long, Any> = ConcurrentHashMap()
+    private val transactionTemplate = TransactionTemplate(transactionManager).apply {
+        propagationBehavior = TransactionDefinition.PROPAGATION_REQUIRES_NEW
```

Comment:
Why not REQUIRED?

Reply:
Because I want to perform this operation in a transaction, that will be completed while lock is acquired. Otherwise, there will be a condition, when different threads access test executions with incorrect statuses.